File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L6/08/Kasus7L6.kt
package IR

import java.util.*

/*
* To change this license header, choose License Headers in Project Properties.
* To change this template file, choose Tools | Templates
* and open the template in the editor.
*/
/**
 *
 * @author B15130F5DDB6B5F1622EF91DAC4C1AAE
 */
object Kasus7L6 {
    @JvmStatic
    fun main(args: Array<String>) {
        //minta input
        val inp = Scanner(System.`in`)
        val matrix = Array(4) { DoubleArray(4) }
        var i = 0
        var sum = 0.0
        print("Enter a 4 by 4 matrix row by row: ")

        //input the matrix
        while (i < 4) {
            var j = 0
            while (j < 4) {
                matrix[i][j] = inp.nextDouble()

                if (i == j) {
                    sum += matrix[i][j]
                }

                j++
            }
            i++
        }

        print("Sum of the elements in the major diagonal is $sum")
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L6/08/Kasus7L6.ts
import { Scanner } from 'readline-sync';

const inp: Scanner = new Scanner();

const matrix: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));
let i: number = 0;
let sum: number = 0;

console.log('Enter a 4 by 4 matrix row by row: ');

while (i < 4) {
  let j: number = 0;
  while (j < 4) {
    matrix[i][j] = inp.nextDouble();
    if (i === j) {
      sum += matrix[i][j];
    }
    j++;
  }
  i++;
}

console.log(`Sum of the elements in the major diagonal is ${sum}`);

